src/System.Diagnostics.Process/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -205,7 +205,14 @@ internal PerformanceMonitor(string machineName)
             private void Init()
             {
 #if FEATURE_REGISTRY
-                _perfDataKey = Registry.PerformanceData;
+                if (_machineName != "." && String.Compare(_machineName, PerformanceCounterLib.ComputerName, StringComparison.OrdinalIgnoreCase) != 0)
```

Comment:
Use`String.Equals(_machineName, PerformanceCounterLib.ComputerName, StringComparison.OrdinalIgnoreCase) `. It's shorter to write, and in the general case slightly faster. Likely not here of course.

Reply:
Actually can't you use the already existing `ProcessManager.IsRemoteMachine(..)` ?

Reply:
Or directly to `IsRemoteMachineCore`